src/interpreter/executors/predicate.rs

```diff
@@ -1,26 +1,28 @@
 use crate::consts::*;
-use crate::error::InterpreterError;
+use crate::error::{InterpreterError, RuntimeError};
+use crate::interpreter::gas::consts::*;
 use crate::interpreter::{Interpreter, MemoryRange};
 use crate::state::{ExecuteState, ProgramState};
-use crate::storage::InterpreterStorage;
 
-impl<S> Interpreter<S>
-where
-    S: InterpreterStorage,
-{
+use fuel_asm::{Immediate18, Instruction, OpcodeRepr, PanicReason, Word};
+
+use std::ops::Div;
+
+impl<S> Interpreter<S> {
     pub(crate) fn verify_predicate(&mut self, predicate: &MemoryRange) -> Result<ProgramState, InterpreterError> {
-        // TODO initialize VM with tx prepared for sign
-        // TODO execute should not overflow predicate boundaries. Need to check
-        // internally if a Jump instruction decrements $pc, or if $pc overflows
-        // `end`
-        let (start, _end) = predicate.boundaries(self);
+        debug_assert!(self.is_predicate());
```

Comment:
This should be an interpreter error imo

Reply:
actually nm, just realized this is internal and only called after the inputs should've been checked